src/components/Project/ImageViewer.vue

```diff
@@ -0,0 +1,210 @@
+
+<script setup>
+import { onMounted, ref } from 'vue'
+import L from 'leaflet'
+import '@geoman-io/leaflet-geoman-free'
+import '@geoman-io/leaflet-geoman-free/dist/leaflet-geoman.css'
+import 'leaflet/dist/leaflet.css'
+// import { fetchApiCall, handleError } from '../../utils/api'
+
+const props = defineProps({
+  imageSrc: {
+    type: String,
+    required: true,
+  }
+})
+
+const isLoading = ref(true)
+
+const imageContainer = ref(null)
+const startCoordinates = ref({x1: 0, y1: 0})
+const endCoordinates = ref({x2: 0, y2: 0})
+let image = null
+let imageOverlay = null
+let imageBounds
+let initialCenter = [0, 0]
+let initialZoom = 1
+let lastDrawnLine = null
+
+// loads image overlay and set bounds
+const loadImageOverlay = (src) => {
+  const img = new Image()
+  img.onload = () => {
+    // Getting image bounds based on img's size
+    imageBounds = [[0, 0], [img.height, img.width]] 
+
+    if (imageOverlay) {
+      image.removeLayer(imageOverlay)
+    }
+    // Add new overlay with correct bounds
+    imageOverlay = L.imageOverlay(src, imageBounds).addTo(image)
+    // Fit image view to the bounds of the image
+    image.fitBounds(imageBounds)
+    image.setMaxBounds(imageBounds)
+
+    initialCenter = image.getCenter()
+    initialZoom = image.getZoom()
+    isLoading.value = false
+  }
+  img.src = src
+}
+
+const displayLineProfile = (data) =>{
+  if (data) {
+    console.log(data)
+  }
+}
+
+async function getLineProfile(startPoint, endPoint) {
+  // const url = 'http://localhost:8000/api/line-profile/'
```

Comment:
Can we add a TODO just so it's clear this commented code is for calling the backend to get the line profile data?

Reply:
yes!